src/Color/Convert.elm

```diff
@@ -133,22 +132,44 @@ Converts a string to `Maybe` of color.
 -}
 hexToColor : String -> Result String Color
 hexToColor =
-    String.toLower
-        >> Regex.find (Regex.AtMost 1)
-            (Regex.regex "^#?([a-f\\d]{2})([a-f\\d]{2})([a-f\\d]{2})$")
-        >> List.map .submatches
-        >> List.head
-        >> Result.fromMaybe "Parsing hex regex failed"
-        >> Result.andThen
-            (\rs ->
-                case List.map (Maybe.map parseIntHex) rs of
-                    (Just (Ok r)) :: (Just (Ok g)) :: (Just (Ok b)) :: [] ->
-                        Ok <| rgb r g b
-
-                    _ ->
-                        -- there could be more descriptive error cases per channel
-                        Err "Parsing ints from hex failed"
-            )
+    let
+        {- Converts "f" to "ff" and "ff" to "ff" -}
+        extend : String -> String
+        extend token =
+            case String.toList token of
+                [ token ] ->
+                    String.fromList [ token, token ]
+
+                _ ->
+                    token
+
+        pattern =
+            ""
+                ++ "^"
+                ++ "#?"
+                ++ "(?:"
+                ++ "(?:([a-f\\d]{2})([a-f\\d]{2})([a-f\\d]{2}))"
+                ++ "|"
+                ++ "(?:([a-f\\d])([a-f\\d])([a-f\\d]))"
+                ++ ")"
+                ++ "$"
+    in
+        String.toLower
+            >> Regex.find (Regex.AtMost 1) (Regex.regex pattern)
+            >> List.head
+            >> Maybe.map .submatches
+            >> Maybe.map (List.filterMap identity)
```

Comment:
There are six groups, three of them are Nothing (3 tokens for short or 3 tokens for full form).